python-backend/app/api/parties/party/resources/manager_resource.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime, timedelta
+import uuid
+
+from flask import request
+from flask_restplus import Resource, reqparse
+from sqlalchemy import or_
+
+from ....mines.mine.models.mine_identity import MineIdentity
+from ..models.party import Party
+from ..models.mgr_appointment import MgrAppointment
+from ....constants import PARTY_STATUS_CODE
+from app.extensions import jwt, api
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+
+class ManagerResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('party_guid', type=str, help='Party guid.')
+    parser.add_argument('mine_guid', type=str, help='Mine guid.')
+    parser.add_argument('effective_date', type=lambda x: datetime.strptime(x, '%Y-%m-%d'), help='Effective date. In the format of YYYY-MM-DD.')
+
+    @api.doc(params={'mgr_appointment_guid': 'Manager guid.'})
+    @jwt.requires_roles(["mds-mine-view"])
+    def get(self, mgr_appointment_guid):
+        manager = MgrAppointment.find_by_mgr_appointment_guid(mgr_appointment_guid)
+        if manager:
+            return manager.json()
+        return self.create_error_payload(404, 'Manager not found'), 404
```

Comment:
This seems backwards. The pattern I'm used to seeing is to check for the error and fall through to the success case

Reply:
I just split the resource into their own file, but fixed